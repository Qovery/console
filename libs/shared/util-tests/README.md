# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
