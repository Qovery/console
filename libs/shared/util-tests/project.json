{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
