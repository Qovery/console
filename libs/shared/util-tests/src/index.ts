export * from './lib/shared-util-tests'
