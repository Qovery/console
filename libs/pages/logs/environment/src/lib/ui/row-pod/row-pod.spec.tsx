import { renderWithProviders } from '@qovery/shared/util-tests'
import RowPod, { type RowPodProps } from './row-pod'

jest.mock('date-fns-tz', () => ({
  format: jest.fn(() => '20 Sept, 19:44:44:44'),
  utcToZonedTime: jest.fn(),
}))

describe('RowPod', () => {
  const props: RowPodProps = {
    index: 1,
    data: {
      created_at: 1667834316521,
      message: 'message',
      pod_name: 'app-z9d11ee4f-7d754477b6-k9sl7',
      version: '53deb16f853aef759b8be84fbeec96e9727',
    },
    filter: [],
    podNameToColor: new Map([['app-z9d11ee4f-7d754477b6-k9sl7', 'rgb(153, 128, 250)']]),
  }

  it('should render successfully', () => {
    const { baseElement } = renderWithProviders(<RowPod {...props} />)
    expect(baseElement).toBeTruthy()
  })

  it('should match snapshot', () => {
    const { container } = renderWithProviders(<RowPod {...props} />)
    expect(container).toMatchSnapshot()
  })
})
