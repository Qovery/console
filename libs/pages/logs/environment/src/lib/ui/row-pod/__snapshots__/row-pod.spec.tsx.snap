// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowPod should match snapshot 1`] = `
<div>
  <div
    class="group flex justify-between font-code text-xs hover:bg-neutral-650 w-full mb-[2px] select-none"
    data-testid="pod-log-row"
  >
    <div
      class="bg-neutral-700 text-neutral-400 group-hover:bg-neutral-550 px-2 pt-0.5 font-code min-w-[40px] max-w-[40px] text-right box-border"
      data-testid="index"
    >
      2
    </div>
    <div
      class="px-4 text-neutral-350 whitespace-nowrap min-w-[225px]"
      data-testid="cell-pod-name"
      style="color: rgb(153, 128, 250);"
    >
      <span
        class="h-5 flex justify-center items-center px-2 bg-neutral-500 rounded-[40px] gap-1"
      >
        app-z9d11e...77b6-k9sl7
        <span
          class="bigger-click-zone cursor-pointer opacity-50"
          data-state="closed"
          data-testid="copy-container"
        >
          <span
            class="icon-solid-copy shrink-0 "
            role="img"
          />
        </span>
      </span>
    </div>
    <div
      class="pt-0.5 flex whitespace-nowrap text-neutral-50 min-w-[85px]"
      data-testid="cell-version"
    >
      <span
        class="group/version"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-code-commit mr-1"
        />
        <span
          class=""
          data-state="closed"
        >
          53deb16
        </span>
        <span
          class="bigger-click-zone cursor-pointer opacity-0 ml-1 group-hover/version:opacity-80"
          data-state="closed"
          data-testid="copy-container"
        >
          <span
            class="icon-solid-copy shrink-0 "
            role="img"
          />
        </span>
      </span>
    </div>
    <div
      class="px-4 pt-0.5 text-neutral-350 whitespace-nowrap"
      data-testid="cell-date"
      title="Mon, 07 Nov 2022 15:18:36 GMT"
    />
    <code
      class="code-ansi select-text pr-6 pt-0.5 text-neutral-50 relative w-full whitespace-pre-wrap break-all"
    >
      <span>
        message
      </span>
    </code>
  </div>
</div>
`;
